meson.build

```diff
@@ -0,0 +1,83 @@
+project('euclidean-rhythms.lv2','c',version: '0.1.0-SNAPSHOT',default_options: ['default_library=shared','c_std=c99'])
+
+# Install folder
+lv2_directory = join_paths(get_option('libdir'), 'lv2')
+install_folder = join_paths(lv2_directory, meson.project_name())
+
+# Sources to compile
+euclidean_rhythms_src = ['plugins/eurhythms.c', 'src/euclidean_algorithms.c']
+
+# Dependencies for euclidean rhythms
+lv2_dep = dependency('lv2', required: true)
+m_dep = meson.get_compiler('c').find_library('m', required: true)
+all_dep = [lv2_dep,m_dep]
+
+# Get the host operating system and cpu architecture
+current_os = host_machine.system()
+current_arch = build_machine.cpu_family()
+
+# Shared c_args for libraries
+lib_c_args = ['-fvisibility=hidden']
+
+# Add default x86 and x86_64 optimizations
+if current_arch == 'x86' or current_arch == 'x86_64' and current_os != 'darwin'
+    lib_c_args += ['-msse','-msse2','-mfpmath=sse','-ffast-math','-fomit-frame-pointer','-fno-finite-math-only']
+endif
+
+# Configure extension for shared object
+if current_os == 'darwin' #mac
+    extension = '.dylib'
+else #unix like (linux, bsd)   
+    extension = '.so'
+endif 
+if current_os == 'windows' #windows
+    extension = '.dll'
+endif
+
+# Build of the shared object
+library('eurhythms',
+    euclidean_rhythms_src,
+    c_args: lib_c_args,
```

Comment:
... because this can just be `gnu_symbol_visibility: 'hidden'` and Meson will handle cross-platform bits for you.